src/tests/unit/tests/common/flux/async-action.test.ts

```diff
@@ -0,0 +1,124 @@
+    });
+
+    test('addListener and invoke with scope', async () => {
+        const scope = 'test_scope';
+        listenerMock.setup(l => l(testPayload)).verifiable(Times.once());
+
+        testObject.addListener(listenerMock.object);
+
+        scopeMutexMock.setup(m => m.tryLockScope(scope)).verifiable(Times.once());
+        scopeMutexMock.setup(m => m.unlockScope(scope)).verifiable(Times.once());
+
+        await testObject.invoke(testPayload, scope);
+    });
+
+    test('invoke calls all registered listeners', async () => {
+        const listenerMocks: IMock<(payload: TestPayload) => Promise<void>>[] = [
+            Mock.ofInstance(_ => Promise.resolve(), MockBehavior.Strict),
+            Mock.ofInstance(_ => Promise.resolve(), MockBehavior.Strict),
+            Mock.ofInstance(_ => Promise.resolve(), MockBehavior.Strict),
+        ];
+
+        listenerMocks.forEach(mock => {
+            mock.setup(l => l(testPayload)).verifiable(Times.once());
+            testObject.addListener(mock.object);
+        });
+
+        scopeMutexMock.setup(m => m.tryLockScope(undefined)).verifiable(Times.once());
+        scopeMutexMock.setup(m => m.unlockScope(undefined)).verifiable(Times.once());
+
+        await testObject.invoke(testPayload);
+
+        listenerMocks.forEach(mock => mock.verifyAll());
+    });
+
+    test('invoke throws if a listener throws error', async () => {
+        const testError = new Error('test error');
+        listenerMock
+            .setup(l => l(testPayload))
+            .returns(async () => {
+                throw testError;
+            })
+            .verifiable(Times.once());
+
+        testObject.addListener(listenerMock.object);
+
+        scopeMutexMock.setup(m => m.tryLockScope(undefined)).verifiable(Times.once());
+        scopeMutexMock.setup(m => m.unlockScope(undefined)).verifiable(Times.once());
+
+        await expect(() => testObject.invoke(testPayload)).rejects.toThrow(testError);
+    });
+
+    test('invoke aggregates errors if multiple listeners throw', async () => {
+        const testErrors: Error[] = [];
+        const listenerMocks: IMock<(payload: TestPayload) => Promise<void>>[] = [
+            Mock.ofInstance(_ => Promise.resolve(), MockBehavior.Strict),
+            Mock.ofInstance(_ => Promise.resolve(), MockBehavior.Strict),
+        ];
+        listenerMocks.forEach(mock => {
+            const testError = new Error('test error');
+            testErrors.push(testError);
+            mock.setup(l => l(testPayload))
+                .returns(async () => {
+                    throw testError;
+                })
+                .verifiable(Times.once());
+            testObject.addListener(mock.object);
+        });
+
+        scopeMutexMock.setup(m => m.tryLockScope(undefined)).verifiable(Times.once());
+        scopeMutexMock.setup(m => m.unlockScope(undefined)).verifiable(Times.once());
+
+        try {
+            await testObject.invoke(testPayload);
```

Comment:
Is it possible to use the `expect(...).toThrowError(...)` syntax here instead of try/ catch?

Reply:
Resolving because this test was removed